### PR TITLE
periph/uart: extend uart API doc

### PR DESCRIPTION
**Pull request by PeterKietzmann**
I guess the title is speaking. Please give feedback about the depth of the description as well as missing informations!

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
I assigned @haukepetersen even if @kaspar030 might be the one to review :-)

File: drivers/include/periph/uart.h [hunk 1]
```diff
@@ -12,6 +12,25 @@
...
+ * module, which automatically enables for receiving data, as well as writing data 
+ * to the UART port, which means transmitting data. The UART device and the 
+ * corresponding pins need to be mapped in `RIOT/boards/* /include/periph_conf.h`. 
+ * Furthermore you need to select the baudrate for initialization which is typically 
+ * {9600, 19200, 38400, 57600, 115200} baud. Additionally you should register a 
+ * callback function that is executed in interrupt context when data is being received.
+ * The driver will then read the received data, call the registered callback function 
+ * and pass the received data to it via its argument.
+ *
+ * By default the `UART_0` device of each board is initialized and mapped to STDIO
```
Comment (haukepetersen) [inline drivers/include/periph/uart.h]: please change to `UART_DEV(0)`

**Comment by haukepetersen** [comment]
I think it makes sense to add the following information:
- we use always the 8N1 Uart configuration (8-bit, no parity, 1 stop bit)
- the interface assumes the RX to be implemented in interrupt driven mode, while TX can be implemented in any way (blocking, interrupt driven, DMA). Rationale: for TX we know how many bytes are going to be send, for RX be mostly don't (so DMA is not an option...)

**Comment by PeterKietzmann** [comment]
@haukepetersen I included some of you comments but tried so stay as "high level" as I could

**Comment by PeterKietzmann** [comment]
Have to leave now. I providently squashed my commits.

**Comment by haukepetersen** [comment]
Travis is not happy, seems like the use of `/*` inside the comment is what causes it.

**Comment by PeterKietzmann** [comment]
Will check

**Comment by PeterKietzmann** [comment]
@haukepetersen no idea what's up with Travis right now, but locally this fixes the build errors. Same for #4504. Are you generally fine with that notation and the rest of the PR(s)?

**Comment by OlegHahm** [comment]
The only thing that is failing currently in Travis is the SQUASH ME commit.

**Comment by PeterKietzmann** [comment]
squashed

**Comment by cgundogan** [comment]
@PeterKietzmann could you also provide a backport to the release branch?

**Comment by haukepetersen** [comment]
ACK and go

**Pull request merged**